src/renderer.ts

```diff
@@ -4,6 +4,7 @@ import { ReactNode, createElement as h, ReactElement } from 'react';
 import { css, jsx as styledH, SerializedStyles } from '@emotion/core';
 import { from, until } from '@guardian/src-foundations/mq';
 import { neutral } from '@guardian/src-foundations/palette';
+import { JSDOM } from 'jsdom';
```

Comment:
Unfortunately, importing `JSDOM` here will cause the rendering code to crash if called on the client (because `JSDOM` only exists server-side). The most straightforward way to solve this problem is to pass down the parser we're currently using (client-side or server-side) as an injected dependency. We're doing this [here](https://github.com/guardian/apps-rendering/blob/d71214098c4697e8de794c8506e4e376b493ca04/src/item.ts#L192) in the parsing code.

On the topic of parsing, I believe by the time we get to the rendering code all HTML should have been parsed out into a data structure (`DocumentFragment`) 🤔. We can then extract information from it using the DOM APIs. Therefore, I think the best place to introduce this is probably in [`parseImage`](https://github.com/guardian/apps-rendering/blob/d71214098c4697e8de794c8506e4e376b493ca04/src/item.ts#L170) within `item.ts`:

```ts
const parseImage = (docParser: DocParser) => (element: BlockElement): Option<Image> => {
  // Some code
  return new Some({
    // Some properties
    caption: docParser(caption),
  });
}
```

while updating the corresponding [`Image` type](https://github.com/guardian/apps-rendering/blob/d71214098c4697e8de794c8506e4e376b493ca04/src/item.ts#L74) to reflect that fact that `caption`s can be more than just a string:

```ts
type Image = {
    kind: ElementKind.Image;
    alt: string;
    caption: DocumentFragment;
    displayCredit: boolean;
    credit: string;
    file: string;
    width: number;
    height: number;
}
```